platform/mk2/hal/cpu_stm32/cpu_device_stm32.c

```diff
@@ -12,7 +13,13 @@
 #define SERIAL_ID_BITS		96
 #define SERIAL_ID_BUFFER_LEN	(((SERIAL_ID_BITS / 8) * 2) + 1)
 
-extern uint32_t _flash_start;
+/*
+ * Set by f407_mem.ld linker script.  Somehow this is getting
+ * altered to a value of 0x20020000.  Don't know why yet.
+ * But we handle this below by masking off the top 12 bits.
+ */
```

Comment:
I'm so confused about this....  Did a quick grep to see if it gets touched anywhere and didn't come up with anything...